Tools/CISupport/ews-build/events.py

```diff
@@ -264,26 +264,38 @@ def _get_commit_msg(self, repo, sha):
 
     @defer.inlineCallbacks
     def _get_pr_files(self, repo, number):
-        # Copied from https://github.com/buildbot/buildbot/blob/v2.10.5/master/buildbot/www/hooks/github.py to include added/modified/deleted
-        headers = {"User-Agent": "Buildbot"}
+        # Heavy modification of https://github.com/buildbot/buildbot/blob/v2.10.5/master/buildbot/www/hooks/github.py
+        PER_PAGE_LIMIT = 100  # GitHub will list a maximum of 100 files in a single response
+        NUM_PAGE_LIMIT = 30  # GitHub stops returning files in a PR after 3000 files
+
+        headers = {}
         if self._token:
             headers["Authorization"] = "token " + self._token
 
-        url = "/repos/{}/pulls/{}/files".format(repo, number)
-        http = yield httpclientservice.HTTPClientService.getService(
-            self.master,
-            self.github_api_endpoint,
-            headers=headers,
-            debug=self.debug,
-            verify=self.verify,
-        )
-        res = yield http.get(url)
-        if 200 <= res.code < 300:
-            data = yield res.json()
-            return [self.file_with_status_sign(f) for f in data]
-
-        log.msg('Failed fetching PR files: response code {}'.format(res.code))
-        return []
+        page = 1
+        files = []
+        while page < NUM_PAGE_LIMIT:
+            response = yield TwistedAdditions.request(
+                url="{}/repos/{}/pulls/{}/files".format(self.github_api_endpoint, repo, number),
+                type=b'GET',
+                params=dict(
+                    per_page=PER_PAGE_LIMIT,
+                    page=page,
+                ), headers=headers,
+                logger=log.msg,
+            )
+            if not response or response.status_code // 100 != 2:
+                break
+            data = response.json()
+            files += [self.file_with_status_sign(f) for f in data]
+            if len(data) == PER_PAGE_LIMIT:
```

Comment:
similar to https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L237, we can reverse this to something like:
```
if len(data) < PER_PAGE_LIMIT:
    break
page += 1
```